Marlin/src/HAL/LPC1768/HAL_SPI.cpp

```diff
@@ -55,27 +55,55 @@
 #include <lpc17xx_pinsel.h>
 #include <lpc17xx_clkpwr.h>
 
+#include "../shared/HAL_SPI.h"
+
+#ifndef HAL_MISO_PIN
```

Comment:
The objective is exactly the opposite... remove “generic” and “un-prefixed” spi configs... not add one more...

Reply:
Things inside a .cpp file are distinct from the rest of the codebase, and this fits what is going on here, while also following a convention.

In general, calling the first exposed SPI port on all boards the "SD SPI Port" is slightly problematic, because it doesn't really apply as an "SD port" unless `SDSUPPORT` is enabled. I expect the vast majority of users will be using this SPI for SD connection primarily, but I'm hesitant to label it as "the SD port."

Reply:
We aren't calling the first spi port as SD spi... we are making the board set the right names for what they mean to.

There's no such a thing as "THE" SPI. This is only valid for old avr that we have only one single spi...

Each device should define its own spi config and pins, and the SD too.

This un-prefixed pins is one of the reasons that all over the code it's wrongly used.

Reply:
@thinkyhead Re-reading your comments, I think here began the misunderstanding of this PR. You said this:

> In general, calling the first exposed SPI port on all boards the "SD SPI Port" is slightly problematic, because it doesn't really apply as an "SD port" unless SDSUPPORT is enabled.

**But this is not true**. The boards only set SPI pins for specific usages. If you look carefully at the board pins, lots of board **doesn't even define SCK/MOSI/MISO when SD is not SPI** (lcd or onboard).

If some DOGM (or another) are sharing SPI configs with SD, this is an error that we are starting to fix!! 

Random boards that doesn't declare MISO/MOSI/SCK pins when the SD is not SPI: `pins_BTT_GTR_V1_0.h, pins_MKS_ROBIN_NANO_V2.h (doesn't declare it at all, because the SD is SDIO), pins_BTT_SKR_common.h (all SKR)`...

It's more than clear the `(MOSI/MISO/SCK/SS)_PIN` are 100% related with SD.